api/handlers/shared/metrics.go

```diff
@@ -3,82 +3,55 @@ package shared
 import (
 	"context"
 	"fmt"
-	"time"
-
-	restError "github.com/kotalco/community-api/pkg/errors"
+	"github.com/gofiber/websocket/v2"
+	"github.com/kotalco/community-api/pkg/k8s"
 	"github.com/kotalco/community-api/pkg/logger"
-	"github.com/kotalco/community-api/pkg/shared"
-	appsv1 "k8s.io/api/apps/v1"
-	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-
-	"github.com/gofiber/websocket/v2"
-	"github.com/kotalco/community-api/pkg/k8s"
-	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/metrics/pkg/apis/metrics/v1beta1"
 )
 
-var metricsClientset = k8s.MetricsClientset()
-
 type metricsResponseDto struct {
 	Cpu    int64 `json:"cpu"`
 	Memory int64 `json:"memory"`
 }
 
+var metricsClientset = k8s.MetricsClientset()
+
 // Metrics returns a websocket that emits cpu and memory usage
 func Metrics(c *websocket.Conn) {
 	defer c.Close()
 
 	name := c.Params("name")
 	ns := c.Locals("namespace").(string)
-	pod := &corev1.Pod{}
 
-	key := types.NamespacedName{
-		Namespace: ns,
-		Name:      fmt.Sprintf("%s-0", name),
-	}
+	// Create a new Metrics Query object to get the CPU and Memory usage of the pod
+	selector := fmt.Sprintf("app.kubernetes.io/managed-by=kotal-operator,app.kubernetes.io/instance=%s", name)
 
-	sts := &appsv1.StatefulSet{}
-	stsKey := types.NamespacedName{
-		Namespace: ns,
-		Name:      name,
+	// Watch the Metrics of the pod using the Metrics API
+	watcher, err := metricsClientset.MetricsV1beta1().PodMetricses(ns).Watch(context.Background(), metav1.ListOptions{
+		LabelSelector: selector,
+	})
+	if err != nil {
+		go logger.Info("METRICS_STREAM", err.Error())
+		return
 	}
 
-podCheck:
-	if err := k8sClient.Get(context.Background(), key, pod); err != nil {
-		go logger.Info("METRICS_POD_NOTFOUND", err.Error())
-		// is the pod error due to sts has been deleted ?
-		stsErr := k8sClient.Get(context.Background(), stsKey, sts)
-		if apierrors.IsNotFound(stsErr) {
-			go logger.Info("METRICS_STS_NOTFOUND", stsErr.Error())
-			c.WriteJSON(shared.NewResponse(restError.NewNotFoundError(stsErr.Error())))
+	for event := range watcher.ResultChan() {
+		if event.Type == "ERROR" {
```

Comment:
why do we need this check ?

Reply:
coz the event could be of type ERROR , 
see https://github.com/kubernetes/apimachinery/blob/master/pkg/watch/watch.go#L67